core/src/main/java/org/axonframework/config/Configurer.java

```diff
@@ -64,6 +64,88 @@
+     * that match the given componentType, unless more specific configuration based on both type and name is available.
+     *
+     * A component matches componentType if componentType is assignable from the component's class. If a component
+     * matches multiple types, and the types derive from each other, the configuration from the most derived type is
+     * used. If the matching types do not derive from each other, the result is unspecified.
+     *
+     * For example: in case a monitor is configured for {@link CommandBus} and another monitor is configured for
+     * {@link org.axonframework.commandhandling.SimpleCommandBus SimpleCommandBus}), components of type
+     * {@link org.axonframework.commandhandling.AsynchronousCommandBus AsynchronousCommandBus} will use the monitor
+     * configured for the SimpleCommandBus.
+     *
+     * @param componentType         The declared type of the component
+     * @param messageMonitorFactory The factory to use
+     * @return the current instance of the Configurer, for chaining purposes
+     */
+    Configurer configureMessageMonitor(Class<?> componentType, MessageMonitorFactory messageMonitorFactory);
+
+    /**
+     * Configures the Message Monitor to use for the Message processing components in this configuration that match
+     * the given class and name.
+     *
+     * A component matches componentType if componentType is assignable from the component's class. If a component
+     * matches multiple types, and the types derive from each other, the configuration from the most derived type is
+     * used. If the matching types do not derive from each other, the result is unspecified.
+     *
+     * For example: in case a monitor is configured for {@link CommandBus} and another monitor is configured for
+     * {@link org.axonframework.commandhandling.SimpleCommandBus SimpleCommandBus}), components of type
+     * {@link org.axonframework.commandhandling.AsynchronousCommandBus AsynchronousCommandBus} will use the monitor
+     * configured for the SimpleCommandBus.
+     *
+     * @param componentType  The declared type of the component
+     * @param componentName  The name of the component
+     * @param messageMonitor The MessageMonitor to use
+     * @return the current instance of the Configurer, for chaining purposes
+     */
+    default Configurer configureMessageMonitor(Class<?> componentType, String componentName, MessageMonitor<Message<?>> messageMonitor) {
+        return configureMessageMonitor(componentType, componentName, (configuration, type, name) -> messageMonitor);
+    }
+
+    /**
+     * Configures the factory create the Message Monitor for those Message processing components in this configuration
+     * that match the given class and name.
+     *
+     * A component matches componentType if componentType is assignable from the component's class. If a component
```

Comment:
Nit: same 'I miss a match-by-name example' as with my previous comment. But again, I'm not sure how important that is.

Reply:
Added a statement in the JavaDoc on name matching.